### ISSUE TITLE
config adds a top-level name key

### ISSUE THREAD

**Issue by janrito** [docker/compose#9355]
**Description**

`docker compose -f docker-compose.yml -f docker-compose.overrides.yml config` will correctly merge the two configurations. But then it will add a top-level `name`.

This key is ok with `docker compose build`, but it fails with other build systems like `docker buildx bake`

**Steps to reproduce the issue:**
1. `docker compose -f docker-compose.yml -f docker-compose.overrides.yml config > docker-compose.resolved.yml`
2. `docker compose -f docker-compose.resolved.yml build` ✅ 
4. `docker buildx bake -f docker-compose.resolved.yml` ❌ 

**Describe the results you received:**

```error: (root) Additional property name is not allowed```
**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker compose version`:**

```
Docker Compose version v2.3.3
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.8.1)
  compose: Docker Compose (Docker Inc., v2.3.3)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 6
  Running: 6
  Paused: 0
  Stopped: 0
 Images: 238
 Server Version: 20.10.13
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2a1d4dbdb2a1030dc5b01e96fb110a9d9f150ecc
 runc version: v1.0.3-0-gf46b6ba
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.104-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 6
 Total Memory: 9.718GiB
 Name: docker-desktop
 ID: 2KES:M5Z5:NZL6:FCHO:P3SG:KOEY:GI4B:ELQL:YCJH:4SGR:4Y5Q:SJ33
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details:**

**Comment by ndeloof** [comment]
need to bump compose-go in buildx

**Comment by ndeloof** [comment]
cc @crazy-max

**Comment by crazy-max** [comment]
@ndeloof Should be already fixed with buildx 0.8.2 right (updated to compose-go 1.2.1 with this release)?: https://github.com/docker/buildx/releases/tag/v0.8.2.

@janrito Can you try with buildx 0.8.2?

**Comment by janrito** [comment]
yep, this is fixed in v0.8.2
